controllers/apply-next/validators.js

```diff
@@ -1,8 +1,33 @@
 'use strict';
-const Joi = require('joi');
 const moment = require('moment');
+const baseJoi = require('joi');
+const { toInteger, isObject } = require('lodash');
+
+const Joi = baseJoi.extend(joi => ({
+    base: joi.date(),
+    name: 'dateObject',
+    coerce: function(value, state, options) {
```

Comment:
This is the thing that lets us submit the date in parts but still validate as a full date object. Uses a joi extension to coerce the nested value into a date string. It's possible to raise errors here through `this.createError`.